users.tf

```diff
@@ -7,6 +7,27 @@ resource "aws_iam_user" "users" {
   name = each.key
 }
 
+# The login profile for each user; note that the user's initial console
+# password is set here, and the user is required to change it at first login.
+resource "aws_iam_user_login_profile" "users" {
+  provider = aws.users
+
+  for_each = toset(keys(var.users))
+
+  password_reset_required = true
+  user                    = aws_iam_user.users[each.key].name
+
+  lifecycle {
+    # Required so that Terraform doesn't reset the password if the user login
+    # profile was created outside of Terraform (password_length) or after the
+    # user has changed their initial password (password_reset_required).
+    ignore_changes = [
```

Comment:
Is this necessary after an initial import? Per the [documentation](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/iam_user_login_profile#argument-reference) they both only apply during resource creation:

> - [`password_length`](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/iam_user_login_profile#password_length-3) - (Optional) The length of the generated password on resource creation. Only applies on resource creation. Drift detection is not possible with this argument. Default value is 20.
> - [`password_reset_required`](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/iam_user_login_profile#password_reset_required-3) - (Optional) Whether the user should be forced to reset the generated password on resource creation. Only applies on resource creation.

Reply:
Good point - let me test that out and make sure it works as advertised.  If so, I will get rid of it after I import the profiles of our previously-created users.

Reply:
I await your update sah.

Reply:
I think we always need it in there.  I tested this in an environment where three of the users have previously set their own passwords.  When I apply with the `lifecycle / ignore_changes` block enabled, Terraform is happy and doesn't want to make any changes.  When I comment out the `lifecycle / ignore_changes` block and run `terraform plan`, it wants to make changes like this to each of those three users:
```
  # aws_iam_user_login_profile.users["test.user"] must be replaced
-/+ resource "aws_iam_user_login_profile" "users" {
      + encrypted_password      = (known after apply)
      ~ id                      = "test.user" -> (known after apply)
      + key_fingerprint         = (known after apply)
      + password                = (known after apply)
      + password_length         = 20 # forces replacement
      ~ password_reset_required = false -> true # forces replacement
        # (1 unchanged attribute hidden)
    }
```

Reply:
Alright, good to know. I'm not sure about the disconnect between the documentation and what we're seeing. Regardless, it looks like the ignores should stay.

Reply:
It looks like some of this may have been fixed in [v5.46.0](https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.46.0). Might be worth revisiting when we move to provider v5.